dobson/main.py

```diff
@@ -65,27 +68,115 @@ def respond_help(slack_client, channel):
         as_user=True,
     )
 
 
+class UserResponse:
+    """
+    A helper class to handle crafting response strings to dobson queries
+    """
+
+    def __init__(self, users, unknown, message, list_all):
+        self.users = users
+        self.unknown = unknown
+        self.message = message
+        self.list_all = list_all
+
+    def random_response(self):
+        """Returns a random response from our response methods"""
+        return random.choice([self.response1])()
+
+    def response1(self):
+        """
+        Sample responses:
+
+        Nobody is at ozone, but there is 1 unknown device...
+        Sean is at ozone, along with the following unknown devices: 000 and 111
+        """
+        # reply containing who is at ozone, e.g. "A, B and C are at ozone"
+        if self.users:
+            reply = '{list_formatted} {is_are} at ozone'.format(**UserResponse.grammar_helper(self.users))
+        else:
+            reply = 'Nobody is at ozone'
+
+        # Add unknown MAC Addresses to the response
+        if self.unknown:
+            unknown_prefix = ', ' + ('along with ' if self.users else ' but there {is_are} ')
+            if self.list_all:
+                unknown_reply = unknown_prefix + 'the following unknown device{s?}: {list_formatted}'
+                unknown_reply = unknown_reply.format(**UserResponse.grammar_helper(self.unknown))
+            else:
+                unknown_reply = unknown_prefix + '{num} unknown device{s?}{ellipsis}'
+                unknown_reply = unknown_reply.format(**UserResponse.grammar_helper(
+                    self.unknown,
+                    ellipsis='...' if not self.users else ''
+                ))
+        else:
+            unknown_reply = '.'
+
+        return reply + unknown_reply
+
+    @staticmethod
+    def grammar_helper(items, **kwargs):
+        """
+        Takes in a list of items and returns a dictionary mapping keywords to strings for use in format strings.
+
+        For example, if we pass in a list with 1 element, the key 'is_are' would be 'is', and a list with 2 or more
+        elements would be 'are'. This enables us to use format strings such as "[list of people] {is_are} home"
+
+        >>> "There {is_are} {num} device{s?} connected".format(**grammar_helper(['sean', 'jason', 'nikhil']))
+        "There are 3 devices connected"
+
+
+        The list of keywords returned:
+        * is_are: either the string 'is' or 'are'
+        * num: the number of elements as a string, e.g. '2'
+        * s?: either the string 's' or the empty string ('')
+        * list_formatted: the output of calling list_to_str on items
+        """
+        return {
+            'is_are': 'is' if len(items) == 1 else 'are',
+            'num': '{:,}'.format(len(items)),
+            's?': '' if len(items) == 1 else 's',
+            'list_formatted': UserResponse.list_to_str(items),
+            **kwargs
+        }
+
+    @staticmethod
+    def list_to_str(lst):
```

Comment:
I really like this, I wish it was built-in somewhere, but I guess it's language-specific :)